### PR TITLE
Feature/convert erd to models and entities

### PR DESCRIPTION
**Pull request by ahmed-ben10**

### CODE REVIEW

File: packages/server/src/app/models/user_quiz/user_quiz.entity.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const { DataTypes, Model } = require('sequelize')
```
Comment (ahmed-ben10) [inline packages/server/src/app/models/user_quiz/user_quiz.entity.js:1]: Make unique constraint:
https://stackoverflow.com/questions/42195348/how-to-define-unique-index-on-multiple-columns-in-sequelize

File: packages/server/src/app/models/user/user.entity.js
Comment (ahmed-ben10) [inline packages/server/src/app/models/user/user.entity.js:1]: add belongs to

Comment (NoNamer777) [reply packages/server/src/app/models/user/user.entity.js:1]: I am not sure what you mean with this comment?

File: packages/server/src/app/models/user_quiz/user_quiz.entity.js (hunk 1 above)
Comment (NoNamer777) [reply packages/server/src/app/models/user_quiz/user_quiz.entity.js:1]: Is this something that still needs doing?

**Review by NoNamer777 — COMMENTED** [review]
Could you maybe have a look at consistent file names for the migration files?
Besides that, the changes look alright

**Review by NoNamer777 — APPROVED** [review]
LGTM 👌🏼

File: packages/server/src/app/models/group/group.entity.js
Comment (majsa1) [inline packages/server/src/app/models/group/group.entity.js:1]: We need the unique code to add users to a group, would it go here?

**Review by majsa1 — CHANGES_REQUESTED** [review]

File: packages/server/src/app/models/group/group.entity.js
Comment (NoNamer777) [reply packages/server/src/app/models/group/group.entity.js:1]: I think so too. could be done in a separate PR too

**Review by NoNamer777 — CHANGES_REQUESTED** [review]
The files regarding the user entity have been moved. I intentionally moved it and moved away from the `models/*` pattern so that there could be a module package for the user related code. This would make it a lot easier to find code and keep code related to each other together.

This we could also apply to other entities like:
- quiz
- badge
- challenge
- group
- etc.

If you agree with that direction, could you please have another look at the code structure that is currently implemented in this branch?

**Review by NoNamer777 — APPROVED** [review]
Nice 💯 👌🏼 🎉

**Pull request merged**